index.bs

```diff
@@ -690,49 +690,46 @@ partial interface MLContext {
 </div>
 
+The {{MLContext/computeSync(graph, inputs, outputs)}} method steps are:
+1. Invoke <a>execute graph</a> given |graph|, |inputs| and |outputs|.
+1. If that throws and error, re-throw the error and stop.
+1. Return {{undefined}}.
+
+To <dfn lt="validate graph resources">validate graph resources</dfn>, given |resources| and |descriptors|, run the following steps:
+1. For each {|key|, |value|} pair of |resources|:
+    1. If |descriptors|[|key|] does not [=map/exist=], return `false`.
+    1. If the type of {{ArrayBufferView}} |value| does not match |descriptors|[|key|].{{MLOperandDescriptor/type}} according to [this table](#appendices-mloperandtype-arraybufferview-compatibility), return `false`.
+    1. If |value|.\[[ByteLength]] is not equal to [=byte length=] of |descriptors|[|key|], return `false`.
+1. Return `true`.
+
+To <dfn lt="execute graph">execute graph</dfn>, given |graph|, |inputs| and |outputs|, run the following steps:
+1. If |graph|.{{MLGraph/[[context]]}}.{{MLContext/[[contextType]]}} is "[=webgpu-context|webgpu=]", throw an "{{OperationError}}" {{DOMException}} and stop.
+1. If invoking the <a>validate graph resources</a> algorithm given |inputs| and |graph|.{{MLGraph/[[inputDescriptors]]}} returns `false`, then throw a "{{DataError}}" {{DOMException}} and stop.
+1. If invoking the <a>validate graph resources</a> algorithm given |outputs| and |graph|.{{MLGraph/[[outputDescriptors]]}} returns `false`, then throw a "{{DataError}}" {{DOMException}} and stop.
+1. To prepare the input resources |inputResources| to the underlying implementation, for each {|key|, |value|} pair of |inputs|:
+    1. Let |inputDescriptor| be |graph|.{{MLGraph/[[inputDescriptors]]}}[|key|].
+    1. Let |inputTensor| be a new tensor for |graph|.{{MLGraph/[[implementation]]}}.
+    1. Set the data type of |inputTensor| to the one that matches the [=element type=] of {{ArrayBufferView}} |value|.
+    1. Set the dimensions of |inputTensor| to |inputDescriptor|.{{MLOperandDescriptor/dimensions}}.
+    1. Set the values of |inputTensor| to the values of |value|.
+    1. Set |inputResources|.|key| to |inputTensor|.
+1. Let |results| be the result of invoking the underlying implementation to compute the output of |graph| given |inputResources|. If that returns an error, throw an "{{OperationError}}" {{DOMException}} and stop.
+1. To validate |results|, for each {|key|, |value|} pair of |outputs|:
+    1. Let |outputTensor| be |results|[key].
+    1. Let |outputDesc| be |graph|.{{MLGraph/[[outputDescriptors]]}}[|key|].
+    1. If the data type of |outputTensor| elements doesn't match the [=element type=] of {{ArrayBufferView}} |value|, then throw a "{{DataError}}" {{DOMException}} and stop.
+    1. If the byte length of |outputTensor| is not equal to |value|.\[[ByteLength]] and the [=byte length=] of |outputDesc|, then throw a "{{DataError}}" {{DOMException}} and stop.
+    1. Otherwise, set the values of |value| to the values of |outputTensor|.
```

Comment:
Possibly improve the prose here for assigning vector elements from another vector.
Is this meant to be a copy, or can it be assignment?